test/unit/controllers/declaration/ConsignmentReferencesControllerSpec.scala

```diff
@@ -63,22 +64,56 @@ class ConsignmentReferencesControllerSpec extends ControllerSpec {
       }
     }
 
-    "return 400 (BAD_REQUEST)" in new SetUp {
+    "return 400 (BAD_REQUEST)" when {
+      "no UCR is submitted" in new SetUp {
 
-      val incorrectForm = Json.toJson(ConsignmentReferences(Some(Ducr("1234")), ""))
+        val incorrectForm = Json.toJson(ConsignmentReferences(Ducr("1234"), ""))
 
-      val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
 
-      status(result) must be(BAD_REQUEST)
+        status(result) must be(BAD_REQUEST)
+      }
+
+      "UCR length has been exceeded" in new SetUp {
+
+        val incorrectForm =
+          Json.toJson(ConsignmentReferences(Ducr("1234"), "", Some(TestHelper.createRandomAlphanumericString(36))))
+
+        val result = controller.submitConsignmentReferences()(postRequest(incorrectForm))
+
+        status(result) must be(BAD_REQUEST)
+      }
+
+      "UCR has special characters" in new SetUp {
```

Comment:
The same here

Reply:
The Form/View Spec is already checking for the personal UCR validation.
This test is just to assert that the HTTP response code, in this scenario, is 400 and not 422, 409 or any other.
A response code needs to be asserted in the controller.

Reply:
How would you test that a 400 response code was replied by the controller?
One needs to mimic a failure scenario, ideally all of them that fall into a `Bad Request`. Please advise ;)

Reply:
Maybe we should provide some way to inject form and at some point in binding return errored one ?

Reply:
but you checked 400 in tests above. This is just another case for form spec. So for Controller spec we just need one situation that will tests case like `if form has error return 400`, but not try to test every case if this value is longer, if this . value has special characters, etc

Reply:
and as I see all tests checks 400 (BAD_REQUEST)

Reply:
if you'll have different spec for 422 or 409 (one for every status) it's everything, but now three specs has 400

Reply:
Ok, it seems to me that there was some thought put into this.
I am going to remove these extra controller tests.
 ;) 
Thanks for your passionate replies.